### PR TITLE
Fast Track: add frontend TypeScript test coverage guardrail

### PR DESCRIPTION
**Pull request by LeonardoRosaa**
## What has changed and why?

This PR implements a new fast-track guardrail that enforces a minimum 80% statement coverage on added lines for frontend TypeScript source files.

## How has it been tested?

Unit tests

## Did you update [CHANGELOG.md](../CHANGELOG.md)?

- [ ] Yes
- [x] Not needed (internal change)

## Summary by CodeRabbit

* **New Features**
  * Added a new frontend coverage check for changed source files.
  * Coverage is now evaluated against added lines, helping flag untested edits more accurately.

* **Bug Fixes**
  * Improved handling of edge cases where coverage data is missing or a file has no relevant test coverage.
  * Better filtering ensures test and declaration files are excluded from coverage checks.

### CODE REVIEW

File: fast_track/src/guardrails/frontend/coverage.ts [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        return Promise.resolve(findFrontendTestFile(sourcePath));
+    },
+
+    async runTests(testFiles: string[], sourcePaths: string[]): Promise<RawCoverage | null> {
+        const relSources = sourcePaths.map((p) => p.slice(FRONTEND_PREFIX.length));
+        const coverageIncludes = relSources.map((f) => `--coverage.include=${f}`);
+        try {
+            await execFileAsync(
+                'npm',
+                ['run', 'test:unit', '--', 'run', '--coverage', ...testFiles, ...coverageIncludes],
```
[the remark this answers is not in the document]
Comment (LeonardoRosaa) [reply fast_track/src/guardrails/frontend/coverage.ts:92]: We should install dependencies in the CI.

**Review by ikondrat — APPROVED** [review]
